### PR TITLE
Minor CMake Changes

### PR DESCRIPTION
**Pull request by dlangbe**
-Changed `AMDGPU_TARGETS`  to `GPU_TARGETS`. `AMDGPU_TARGETS` will be deprecated in the future, but will work the same for the time being.
-Removed the use of `CMAKE_NO_BUILTIN_CHRPATH` when --offload-compress is supported.

### CODE REVIEW

**Comment by amd-jnovotny** [comment]
@dlangbe Docs changes look good, but do we need to update the changelog here too? (We did for hipTensor 303.)

**Comment by cgmillette** [comment]
@dlangbe We should apply the feedback from hipTensor's PR to this one. Also need to target as ROCm 6.4 change

**Review by amd-jnovotny — APPROVED** [review]
looks good

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Documentation for rocWMMA is available at
 [https://rocm.docs.amd.com/projects/rocWMMA/en/latest](https://rocm.docs.amd.com/projects/rocWMMA/en/latest).
 
+## (Unreleased) rocWMMA 1.7.0 for ROCm 6.4.0
+
+### Changed
+
+* Renamed the CMake option `AMDGPU_TARGETS` to `GPU_TARGETS`
+
 ## (Unreleased) rocWMMA 1.6.0 for ROCm 6.3.0
```
Comment (cgmillette) [inline CHANGELOG.md]:
```suggestion
## rocWMMA 1.6.0 for ROCm 6.3.0
```

File: README.md [hunk 2]
```diff
@@ -47,7 +47,8 @@ For more detailed information, please refer to the [rocWMMA installation guide](
 
 |Option|Description|Default value|
 |---|---|---|
-|AMDGPU_TARGETS|Build code for specific GPU target(s)|gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1100;gfx1101;gfx1102|
+|GPU_TARGETS|Build code for specific GPU target(s)|gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1100;gfx1101;gfx1102|
```
Comment (cgmillette) [inline README.md]:
```suggestion
|GPU_TARGETS|Build code for specific GPU target(s)|gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1100;gfx1101;gfx1102;gfx1200;gfx1201|
```

File: README.md [hunk 3]
```diff
@@ -47,7 +47,8 @@ For more detailed information, please refer to the [rocWMMA installation guide](
...
+|AMDGPU_TARGETS|(Deprecated) Build code for specific GPU target(s)|gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1100;gfx1101;gfx1102|
```
Comment (cgmillette) [inline README.md]:
```suggestion
|AMDGPU_TARGETS|(Deprecated) Build code for specific GPU target(s)|gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+;gfx1100;gfx1101;gfx1102;gfx1200;gfx1201|
```

File: docs/install/installation.rst [hunk 4]
```diff
@@ -183,7 +183,7 @@ Below are the project options available to build rocWMMA library with or without
     *   -   **Option**
         -   **Description**
         -   **Default Value**
-    *   -   AMDGPU_TARGETS
+    *   -   GPU_TARGETS
         -   Build code for specific GPU target(s)
         -   ``gfx908:xnack-``; ``gfx90a:xnack-``; ``gfx90a:xnack+``; ``gfx940``; ``gfx941``; ``gfx942``; ``gfx1100``; ``gfx1101``; ``gfx1102``
```
Comment (cgmillette) [inline docs/install/installation.rst]:
```suggestion
        -   ``gfx908:xnack-``; ``gfx90a:xnack-``; ``gfx90a:xnack+``; ``gfx940``; ``gfx941``; ``gfx942``; ``gfx1100``; ``gfx1101``; ``gfx1102``;``gfx1200``;``gfx1201``
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -96,28 +96,40 @@ endif()
 
 if (ADDRESS_SANITIZER_ENABLED)
     #TODO: Remove next line when rocm-cmake fix is available
-    set(CMAKE_NO_BUILTIN_CHRPATH ON)
-    rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
+    rocm_check_target_ids(DEFAULT_GPU_TARGETS
         TARGETS "gfx90a:xnack+;gfx942:xnack+" )
 else()
     #TODO: Remove next line when rocm-cmake fix is available
```
Comment (cgmillette) [inline CMakeLists.txt]:
```suggestion
```

File: CMakeLists.txt [hunk 6]
```diff
@@ -96,28 +96,40 @@ endif()
...
+    rocm_check_target_ids(DEFAULT_GPU_TARGETS
         TARGETS "gfx908;gfx90a;gfx942;gfx1100;gfx1101;gfx1102;gfx1200;gfx1201" )
 endif()
 
+# Check if offload compression is supported
+include(CheckCXXCompilerFlag)
+if (BUILD_OFFLOAD_COMPRESS)
+  check_cxx_compiler_flag("--offload-compress" CXX_COMPILER_SUPPORTS_OFFLOAD_COMPRESS)
+  if (NOT CXX_COMPILER_SUPPORTS_OFFLOAD_COMPRESS)
+    set(CMAKE_NO_BUILTIN_CHRPATH ON)
```
Comment (cgmillette) [inline CMakeLists.txt:112]:
```suggestion
    #TODO: Remove next line when rocm-cmake fix is available
    set(CMAKE_NO_BUILTIN_CHRPATH ON)
```

**Review by cgmillette — APPROVED** [review]

File: test/CMakeLists.txt [hunk 7]
```diff
@@ -88,6 +88,10 @@ function(add_rocwmma_test TEST_TARGET TEST_SOURCE)
   target_include_directories(${TEST_TARGET} PRIVATE
                              ${CMAKE_CURRENT_SOURCE_DIR}
                              ${ROCWMMA_TEST_INCLUDE_DIRS})
+  # Use offload compress flag if it is supported
+  if(BUILD_OFFLOAD_COMPRESS AND CXX_COMPILER_SUPPORTS_OFFLOAD_COMPRESS)
+    target_compile_options(${COMPONENT_NAME} PRIVATE "--offload-compress" )
```
Comment (dlangbe) [inline test/CMakeLists.txt]:
```suggestion
    target_compile_options(${TEST_TARGET} PRIVATE "--offload-compress" )
```

**Pull request merged**